ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.html

```diff
@@ -1,27 +1,51 @@
+<!--  TODO: Add i18n  -->
 <ddp-question-prompt [block]="block"></ddp-question-prompt>
-<div class="file-upload-content">
+
+<div class="file-requirements" *ngIf="fileMaxSize || fileMimeTypes.length">
+    <div class="file-max-size" *ngIf="fileMaxSize">Max size: {{fileMaxSize}}</div>
+    <div class="file-types" *ngIf="fileMimeTypes.length">Allowed mime-types: {{fileMimeTypes.join(', ')}}</div>
+</div>
+
+<div class="file-upload-content" [class.readonly]="readonly">
     <input type="file" class="file-input"
+           [accept]="fileMimeTypes.join(',')"
            (change)="onFilesSelected($event.target.files)" #fileUpload>
 
     <div class="file-select">
-        <div class="drop-block" dropFileToUpload (filesDropped)="onFilesSelected($event)">
+        <div class="drop-block"
+             dropFileToUpload
+             (filesDropped)="!readonly && onFilesSelected($event)">
             <mat-icon class="upload-icon" color="primary">file_upload</mat-icon>
             <span class="drop-block-text">Drag files to upload</span>
         </div>
         <button mat-raised-button color="primary" class="upload-btn"
+                [disabled]="readonly"
                 (click)="fileUpload.click()">Choose file
             <mat-icon>attach_file</mat-icon>
         </button>
 
+        <div class="selected-file">{{selectedFile?.name}}</div>
+
+        <button *ngIf="!readonly" #uploadBtn
+                mat-raised-button color="primary"
+                class="submit-btn"
+                [disabled]="!isFileSelected"
+                (click)="submitFileUpload()">{{uploadedFile?.isUploaded? 'Reupload' : 'Submit'}}
+        </button>
     </div>
-    <div class="file-info">
-        <span class="file-name">{{file?.name || 'No file uploaded'}}</span>
+
+    <div class="uploaded-file">
+        <div class="title">Uploaded:</div>
+
+        <mat-chip-list>
+            <mat-chip *ngIf="uploadedFile?.isUploaded"
+                      class="uploaded-file-chip"
+                      color="primary" selected
+                      (removed)="undoUploadedFile()">{{uploadedFile.name}} (size: {{uploadedFile.size}})
+                <mat-icon matChipRemove *ngIf="!readonly" #undoUploadBtn>cancel</mat-icon>
```

Comment:
We can actually tell the server to delete the upload if they press the cancel button.
The way to do this is to PATCH an answer with a null value.
So right now, after the client uploads the contents of the file, your code correctly issues a PATCH that looks like:
```json
{"answers":[{"stableId":"FILE_UPLOAD_1","answerGuid":"SM27JVPJDB","value":"2kn3N8lkJXzmh12sFEMj"}]}
```
To delete the upload (i.e., when the matChipRemove is clicked) we would want to PATCH:
```json
{"answers":[{"stableId":"FILE_UPLOAD_1","answerGuid":"SM27JVPJDB","value":null}]}

Reply:
Right. It should work exactly like you wrote.
For example:
1.Upload a file
![upload1](https://user-images.githubusercontent.com/7396837/119811332-a767f000-beef-11eb-9187-9c0a08d8ba88.png)

2. Deleting the already uploaded file
![upload2](https://user-images.githubusercontent.com/7396837/119811395-b8186600-beef-11eb-8619-00c4b0f9b7e8.png)